mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -1266,16 +1272,18 @@ private Query createQuery(MessageRange range) {
     }
 
     @Override
-    public void delete(MailboxSession session, Mailbox mailbox, Collection<MessageUid> expungedUids) throws IOException {
-        Collection<MessageRange> messageRanges = MessageRange.toRanges(expungedUids);
-        for (MessageRange messageRange : messageRanges) {
-            delete(mailbox.getMailboxId(), messageRange);
-        }
+    public Mono<Void> delete(MailboxSession session, Mailbox mailbox, Collection<MessageUid> expungedUids) {
+        return Mono.fromRunnable(Throwing.runnable(() -> {
+            Collection<MessageRange> messageRanges = MessageRange.toRanges(expungedUids);
+            for (MessageRange messageRange : messageRanges) {
```

Comment:
idem ?

Reply:
That would bring little value IMO as the underlying steps are fully synchronous.

Feel free to contribute it once this work is merged.